src/components/ExampleDisplayComponent.js

```diff
@@ -0,0 +1,131 @@
+CodeMirror.defineSimpleMode('fsh', {
+  meta: {
+    lineComment: '//'
+  },
+  start: [
+    // The regex matches the token, the token property contains the type
+    {
+      regex: /"(?:[^\\]|\\.)*?(?:"|$)/,
+      token: 'atom'
+    },
+    {
+      regex: /\b(Alias|CodeSystem|Expression|Extension|Description|Id|Instance|InstanceOf|Invariant|Mapping|Mixins|Parent|Profile|RuleSet|Severity|Source|Target|Title|Usage|ValueSet|XPath)(?=\s*:)\b/,
+      token: 'keyword'
+    },
+    {
+      // NOTE: Original regex has (?<=\s) at start and (?=\s) at the end. However, there are known shortcomings with look ahead/look behind with the simple mode approach
+      regex: /\b(and|codes|contains|descendent-of|exclude|exists|from|generalizes|include|in|insert|is-a|is-not-a|named|not-in|obeys|only|or|regex|system|units|valueset|where|D|MS|N|SU|TU|\\?!)\b/,
+      token: 'def'
+    },
+    {
+      regex: /(\(\s*)(exactly|example|extensible|preferred|required)(\s*\))/,
+      token: 'def'
+    },
+    {
+      regex: /\*|->|=|:/,
+      token: 'def'
+    },
+    {
+      regex: /\b(https?:\/\/(www\.)?[-a-zA-Z0-9@:%._+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_+.~#?&//=]*))\b/,
+      token: 'atom'
+    },
+    {
+      regex: /\b(true|false)\b/,
+      token: 'string'
+    },
+    {
+      regex: /#[^\s]*/,
+      token: 'string'
+    },
+    { regex: /\/\/.*/, token: 'comment' },
+    { regex: /\/\*/, token: 'comment', next: 'comment' }
+  ],
+  comment: [
+    { regex: /.*?\*\//, token: 'comment', next: 'start' },
+    { regex: /.*/, token: 'comment' }
+  ]
+});
+
+const useStyles = makeStyles((theme) => ({
+  box: {
+    height: '100%'
+  },
+  header: {
+    fontFamily: 'Open Sans',
+    color: theme.palette.common.white,
+    background: '#424242', // Dark mode background
+    padding: '0px',
+    paddingLeft: '29px', // width of code mirror gutter
+    height: '34px', // 24px + 10px of padding is total height
+    transition: theme.transitions.create(['margin', 'width'], {
+      easing: theme.transitions.easing.sharp,
+      duration: theme.transitions.duration.leavingScreen
+    })
+  },
+  headerLabel: {
+    lineHeight: '34px',
+    float: 'left'
+  }
+}));
+
+export default function CodeMirrorComponent(props) {
+  const classes = useStyles();
+
+  //Updates both the text state with the codemirror value
+  function updateText(text) {
+    props.updateTextValue(text);
+  }
+
+  return (
+    <Box className={classes.box}>
+      <div className={clsx(classes.header)}>
+        <div className={classes.headerLabel}>{props.name}</div>
+      </div>
+      <ReactCodeMirror
+        className={clsx('react-codemirror2')}
+        value={props.initialText}
+        options={{
+          mode: props.mode,
+          theme: 'material',
+          placeholder: props.placeholder,
+          autoCloseBrackets: true,
+          lineNumbers: true,
+          foldGutter: true,
+          readOnly: true,
```

Comment:
I actually left that in as a feature! Mark thought it would be useful for users to be able to pop into an example and copy a portion of it to paste into the main editor. I thought that allowing them to edit the example in the actual display was a bit problematic though, so I made the component read only.

Reply:
Ohh gotcha! If that's what lets you copy just part of an example, then I like it! I definitely tried to do that and it worked nicely. I just also saw the cursor and thought I would be able to type, even though I agree that you _shouldn't_ be able to type.